src/main/java/nl/tudelft/goalkeeper/parser/results/files/module/parsers/ModuleParser.java

```diff
@@ -26,9 +34,54 @@ public static ModuleFile parseToFile(languageTools.program.agent.Module m) throw
         ModuleFile module = new ModuleFile(m.getSourceFile().toString());
         addRules(module, m);
         module.setName(m.getName());
+        module.setKRLanguage(getKRLanguage(module.getRules()));
+        setDetails(module, m);
         return module;
     }
 
+    /**
+     * Finds the KRLanguage of a set of rules.
+     * @param rules List of rules to find the KRLanguage of.
+     * @return The KRLanguage of the rules list.
+     */
+    private static KRLanguage getKRLanguage(List<Rule> rules) {
+        for (Rule rule : rules) {
+            if (rule.getKRLanguage() != KRLanguage.UNKNOWN) {
+                return rule.getKRLanguage();
```

Comment:
Could be, but shouldn't be, since GOAL modules can only have one KR language at a time. So unless you manually recreate that scenario it should be fine.

Reply:
Ah okee, now I am wondering if you can have several KRLanguages within a goal project or if it is just module specific?

Reply:
Module specific AFAIK, although I'm not sure how that would work

Reply:
ah okee me neither unfortunately